cache.rst

```diff
@@ -0,0 +1,192 @@
+- :ref:`PHPFileAdapter <component-cache-php-files-adapter>`
+- :ref:`ProxyAdapter <component-cache-proxy-adapter>`
+- :ref:`cache.adapter.redis <component-cache-redis-adapter>`
+- cache.adapter.psr6
+- cache.adapter.system
+- NullAdapter
+- cache.adapter.array
+
+Some of these adapters could be configured via shortcuts:
+
+.. code-block:: yaml
+
+    framework:
+        cache:
+            directory: '%kernel.cache_dir%/pools' # Only used with cache.adapter.filesystem
+
+            default_doctrine_provider: 'app.doctrine_cache'
+            default_psr6_provider: 'app.my_psr6_service'
+            default_redis_provider: 'redis://localhost'
+            default_memcached_provider: 'memcached://localhost'
+            default_pdo_provider: 'doctrine.dbal.default_connection'
+
+Using the *providers* above, they will configure the adapters and create pools (services)
+with service ids:
+
+- ``cache.doctrine``
+- ``cache.psr6``
+- ``cache.redis``
+- ``cache.memcached``
+- ``cache.pdo``
+
+Creating Custom Pools
+---------------------
+
+You can also create more customized pools. All you need is an adapter
+
+.. code-block:: yaml
+
+    framework:
+        cache:
+            default_memcached_provider: 'memcached://localhost'
+            pools:
+                my_cache_pool:
+                    adapter: cache.adapter.array
+                cache.acme:
+                    adapter: cache.adapter.memcached
+                cache.foobar:
+                    adapter: cache.adapter.memcached
+                    provider: 'memcached://user:password@example.com'
+
+The configuration above will create 3 services: ``my_cache_pool``, ``cache.acme``
+and ``cache.foobar``.  The ``my_cache_pool`` pool is using the ArrayAdapter
+and the other two are using the :ref:`MemcachedAdapter <component-cache-memcached-adapter>`.
+The ``cache.acme`` pool is using the Memcached server on localhost and ``cache.foobar``
+is using the Memcached server at example.com.
+
+For advanced configurations it could sometimes be useful to use a pool as an adapter.
+
+.. code-block:: yaml
+
+    framework:
+        cache:
+            pools:
+                my_cache_pool:
+                    adapter: cache.adapter.memcached
+                    provider: 'memcached://user:password@example.com'
+                cache.short_cache:
+                    adapter: my_cache_pool
+                    default_lifetime: 60
+                cache.long_cache:
+                    adapter: my_cache_pool
+                    default_lifetime: 604800
+
+Creating a Cache Chain
+----------------------
+
+Different cache adapters has different strengths and weaknesses. Some might be really
+quick but small and some may be able to contain a lot of data but are quite slow.
+To get the best of both worlds you may use a chain of adapters. The idea is to
+first look at the quick adapter and then move on to slower adapters. In the worst
+case we have to recalculate the value.
```

Comment:
Please always avoid first-person usage in the docs, i.e: "In the worst case, the value needs to be recalculated"